src/common/components/ChangeLog.tsx

```diff
@@ -0,0 +1,160 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { IconProp } from '@fortawesome/fontawesome-svg-core';
+import { FC } from 'react';
+import { Button, ListGroup } from 'react-bootstrap';
+import { CircularContainer } from 'common/styles/utilities';
+import styled from 'styled-components';
+import { isKeyOfObject, isObject, isStringOrNumber } from 'common/error/utilities';
+
+export type Props = {
+  changeList: unknown[];
+  totalChanges: number;
+  changePropertyAccessor?: (change: unknown, propertyName: string) => string | number;
+  getChangeItemAction?: (action: string) => string;
+  handleShowAllChanges: () => void;
+};
+
+const StyledCircularContainer = styled(CircularContainer)`
+  background-color: lightgrey;
+  height: 32px;
+  width: 32px;
+`;
+
+const StyledEditorName = styled.span`
+  color: ${props => props.theme.changelogs.accentTextColor};
+`;
+
+const StyledShowAllChangesButton = styled(Button)`
+  color: ${props => props.theme.changelogs.accentTextColor};
+`;
+
+const defaultPropertyAccessor = (
+  change: unknown,
+  propertyName: string,
+): string | number | Record<string, unknown> | boolean => {
+  if (isObject(change)) {
+    const value = change[propertyName];
+
+    if (isStringOrNumber(value)) {
+      return value;
+    }
+
+    if (isObject(value)) {
+      return value;
+    }
+
+    if (value === null) {
+      return false;
+    }
+
+    throw Error(`The resulting value from change[${propertyName}] is not a string, number, object, or null`);
+  }
+
+  throw Error('Given change parameter is not an object');
+};
+
+const defaultGetChangeItemAction = (action: string): string => {
+  switch (action) {
+    case '+':
+      return 'create';
+    case '~':
+      return 'update';
+    case '-':
+      return 'delete';
+    default:
+      throw Error("Given action does not equal '+', '~', or '-'");
```

Comment:
We are mapping these to `create`, `update` and `delete` only to transform them later into `created`, `updated` and `deleted` later. Maybe it's best here to just used the past tense names, only since they are the ones that are displayed.